docs/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,219 @@
+      console.log(collection); // Collection with all default fields and products with all default fields.
+    });
+
+    // or specify a custom query
+    const query = collectionNodeQuery(['title', ['products', productConnectionQuery(['title'])]]);
+
+    client.fetchCollection(collectionId, query).then((collection) => {
+      console.log(collection); // Collection with only title and products with only title
+    });
+    ```
+    Note that `fetchCollectionWithProducts()` and `fetchCollectionsWithProducts()` do not take a `query` argument.
+
+4.  `fetchQueryProducts()` and `fetchQueryCollections()` query different fields and take an optional `query` argument.
+    See the [API reference](https://github.com/Shopify/js-buy-sdk/blob/v1.0alpha/docs/API_REFERENCE.md) for more details.
+
+### Carts/Checkouts
+
+Carts are replaced with checkouts. Like the fetch functions, all checkout functions take an optional `query` argument that specifies fields to return on the checkout.
+
+`fetchRecentCart()`, `updateModel()`, and `clearLineItems()` have been removed. `checkoutUrl` has been renamed to `webUrl`.
+
+#### Creating a Checkout
+
+To create a checkout, use `createCheckout()`.
+
+**v0.7:**
+```js
+client.createCart().then((cart) => {
+  console.log(cart); // Empty cart
+});
+```
+
+**v1:**
+```js
+client.createCheckout().then((checkout) => {
+  console.log(checkout); // Empty checkout
+});
+```
+The checkout can also be initialized with fields like line items and a shipping address. See the [API reference](https://github.com/Shopify/js-buy-sdk/blob/v1.0alpha/docs/API_REFERENCE.md#Client+createCheckout) for more details. 
+
+#### Fetching a Checkout
+
+To fetch a checkout, use `fetchCheckout()`.
+
+**v0.7:**
+```js
+client.fetchRecentCart().then((cart) => {
+  console.log(cart); // Most recently created cart
+});
+
+// or
+client.fetchCart('shopify-buy.1459804699118.2').then(cart => {
+  console.log(cart); // The retrieved cart
+});
+```
+
+**v1:**
+```js
+const checkoutId = 'Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0SW1hZ2UvMTgyMTc3ODc1OTI=';
+
+client.fetchCheckout(checkoutId).then((checkout) => {
+  console.log(checkout); // The retrieved checkout
+});
+```
+#### Modifying an Existing Checkout
+
+The functions to modify a checkout are on `Client` rather than `CartModel`.
+
+##### Adding Line Items
+
+To add line items to a checkout, use `addLineItems()` (previously `createLineItemsFromVariants()`).
+
+**v0.7:**
+```js
+cart.createLineItemsFromVariants({variant: variantObject1, quantity: 5}, {variant: variantObject2, quantity: 2}).then((cart) => {
+  console.log(cart); // Cart with two additional line items
+});
+```
+
+**v1:**
+```js
+const checkoutId = 'Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0SW1hZ2UvMTgyMTc3ODc1OTI=';
+const lineItems = [
+  {variantId: 'Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0VmFyaWFudC8yNTYwMjIzNTk3Ng==', quantity: 5},
+  {variantId: 'Z2lkOi8vc2hvcGlmeS9Qcm9kdWN0VmFyaWFudC8yNTYwMjIzNjA0MA==', quantity: 2}
```

Comment:
since there's a new feature around line item attributes, maybe show a second example in here about how to pass them in?